### PR TITLE
Couple of memory leak fixes

### PR DESCRIPTION
**Pull request by devnexen**
Delete where appropriate 

## Original issue

Link here

## Description

Write here

## Check List

* [ ] Tests written (if appropriate)
* [x] Locally tested
* [ ] Documented well enough

### CODE REVIEW

File: source/src/main.cpp [hunk 1]
```diff
@@ -215,6 +215,7 @@ void bmp_screenshot(const char *imagepath, bool mapshot = false)
         else
         {
             conoutf("no mapshot prepared!");
+            delete tmp;
```
Comment (lucasgautheron) [inline source/src/main.cpp]: "tmp" is an array of pointers here. It should therefore be freed with the proper c++ operator for this, "delete[]".
But in this case, prefer to use DELETEA() (a custom-defined macro) over it. If any of its instructions are unneeded, they should be optimized out anyway.

File: source/src/main.cpp [hunk 2]
```diff
@@ -440,6 +441,7 @@ void png_screenshot(const char *imagepath, bool mapshot = false)
...
```
Comment (lucasgautheron) [inline source/src/main.cpp]: Same comment as above.

File: source/src/stream.cpp [hunk 3]
```diff
@@ -924,7 +924,7 @@ stream *opengzfile(const char *filename, const char *mode, stream *file, int lev
     stream *source = file ? file : openfile(filename, mode, allow_outside_game);
     if(!source) return NULL;
     gzstream *gz = new gzstream;
-    if(!gz->open(source, mode, !file, level)) { if(!file) delete source; return NULL; }
+    if(!gz->open(source, mode, !file, level)) { if(!file) delete source; delete gz; return NULL; }
```
Comment (lucasgautheron) [inline source/src/stream.cpp]: Please use our macro DELETEP() to free regular pointers.

File: source/src/main.cpp (hunk 1 above)
Comment (devnexen) [inline source/src/main.cpp]: good catch did not notice there were arrays.

**Comment by lucasgautheron** [comment]
Thanks !

**Pull request merged**